src/main/kotlin/com/sourcegraph/cody/config/RateLimitStateManager.kt

```diff
@@ -18,6 +19,7 @@ object RateLimitStateManager {
     if (UpgradeToCodyProNotification.chatRateLimitError.get() == null) {
       UpgradeToCodyProNotification.chatRateLimitError.set(rateLimitError)
       CodyAutocompleteStatusService.resetApplication(project)
+      CodyToolWindowContent.getInstance(project).refreshSubscriptionTab()
```

Comment:
adding this method here makes `reportForChat` `@RequiresBackgroundThread`. pls annotate it and ensure that it's called in the proper thread.

Reply:
you can add `Thread.sleep(10000)` and see if the UI remains functional